apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/class-wpcom-block-editor-nux.php

```diff
@@ -75,7 +75,7 @@ public function enqueue_script_and_style() {
 	 * Register the WPCOM Block Editor NUX endpoints.
 	 */
 	public function register_rest_api() {
-		require_once __DIR__ . '/class-wp-rest-wpcom-block-editor-nux-status-controller.php';
+		require_once __DIR__ . '/../wpcom-block-editor-welcome-tour/class-wp-rest-wpcom-block-editor-nux-status-controller.php';
 		$controller = new WP_REST_WPCOM_Block_Editor_NUX_Status_Controller();
 		$controller->register_rest_route();
```

Comment:
I forgot to add: because the other usage of this controller also attempts to register the route, we might just want to test that this doesn’t throw an error anywhere. I think you can safely call to register the same route twice, but we might want to double check (check that it doesn’t throw errors and that the endpoint behaves as we’d expect). The function register_rest_route says that it returns false if it can register an endpoint so I suspect it’ll safely silently fail the second time, or just override the first call, which is also fine 🤞

Reply:
@andrewserong Thanks so much for pointing this out!  Got me to think/learn a bit more in depth about some WP & php concepts.

> register the same route twice

If I understand you correctly, the concern is that `
class WP_REST_WPCOM_Block_Editor_NUX_Status_Controller` will be registered twice, once when NUX code loads and again when Tour code loads.   However, only one of these will load at a time, it is an either/or situation for A/B testing.  This [code in full-site-editing-plugin.php](https://github.com/Automattic/wp-calypso/blob/137073582964d86579a57c9f85b63ed70b0af831/apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php#L243) decides which to load.

In my sandbox I tested both scenarios and I didn't observe any errors in my `/tmp/php-errors` file or in dev tools. (Note: Adding `define( 'SHOW_WELCOME_TOUR', true );` to 0-sandbox.php file will force the new Tour to show). 

Does that address the issue? Or is there anything I'm missing/not understanding?

Reply:
@autumnfjeld perfect, that sounds good to me! And from reading [the docs](https://developer.wordpress.org/reference/functions/register_rest_route/) for `register_rest_route`, it looks like even if it wasn't guarded behind that either/or condition we'd probably still be safe 👍